src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -124,64 +124,59 @@ def acr_show(registry_name, resource_group_name=None):
 
     return client.get_properties(resource_group_name, registry_name)
 
-def acr_update(registry_name, #pylint: disable=too-many-arguments
-               resource_group_name=None,
-               storage_account_name=None,
-               tags=None,
-               enable_admin=False,
-               disable_admin=False):
-    '''Update a container registry.
-    :param str registry_name: The name of container registry
-    :param str resource_group_name: The name of resource group
-    :param str storage_account_name: The name of storage account
-    :param dict tags: The set of tags
-    :param bool enable_admin: Enable admin user
-    :param bool disable_admin: Disable admin user
-    '''
-    registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
-
-    # Set admin_user_enabled
-    admin_user_enabled = None
-    if disable_admin:
-        admin_user_enabled = False
-    if enable_admin:
-        admin_user_enabled = True
-
-    # Set tags
-    newTags = registry.tags #pylint: disable=no-member
-    if isinstance(tags, dict):
-        if tags:
-            for key in tags:
-                if tags[key]:
-                    newTags[key] = tags[key]
-                elif key in newTags:
-                    del newTags[key]
-        else:
-            newTags = {}
-
-    # Set storage account
-    storage_account = None
-    if storage_account_name:
-        storage_account_key = get_access_key_by_storage_account_name(storage_account_name)
+def acr_update_get(client,
+                      registry_name,
+                      resource_group_name=None):
+    if resource_group_name is None:
+        resource_group_name = get_resource_group_name_by_registry_name(registry_name)
+
+    props = client.get_properties(resource_group_name, registry_name)
+
+    return RegistryUpdateParameters(
+        tags=props.tags,
+        admin_user_enabled=props.admin_user_enabled,
+        storage_account=props.storage_account
+    )
+
+def acr_update_custom(instance,
+                      admin_user_enabled=None,
+                      storage_account_name=None,
+                      tags=None):
+    if admin_user_enabled is not None:
+        instance.admin_user_enabled = admin_user_enabled == 'true'
```

Comment:
If you use `action='store_true'`, this would just be a simple boolean instead of a string comparison to 'true'.